src/main/webapp/script.js

```diff
@@ -29,30 +29,60 @@ function loadTasks() {
 }
 
 function renderTasks(tasks) {
-  // TODO: check the size of tasks and show a warning if zero
-  $("#task-list").empty();
-  tasks.forEach(renderSingleTask);
+  const taskList = $("#task-list");
+  taskList.empty();
+  if (tasks.length > 0) {
+    $("#schedule-button").prop("disabled", false);
+    tasks.forEach(renderSingleTask);
+  } else {
+    $("#schedule-button").prop("disabled", true);
+    taskList.html(
+      `<li class="mdc-list-item default-cursor">
+        <span class="mdc-list-item__text">There are no tasks to schedule</span>
+       </li>`);
+  }
 }
 
 function renderSingleTask(task) {
   id = task.id;
-  newTask = $("<p></p>", {"id": id});
-  checkBox = $("<input>", {"name": "taskId", "value": id, "type": "checkbox"});
-  checkBox.appendTo(newTask);
-  newTask.append(task.title);
-  newTask.addClass("task");
-  newTask.appendTo("#task-list");
+  $("#task-list").append(
```

Comment:
Similar thing with HTML in the script.
Suggested solution (feel free to ignore, because it is quite complex):
1. Move this HTML into the index.html, make it hidden (e.g. putting  `style="display: none;">`) and remove all dynamic values.
2. At the beginning of the script, get the top-level node via `$()` and detach it from the DOM, remove hidden style.
3. Whenever you would like to create a new task entry here, you'd `.clone()` the saved node, set values and attributes for all dynamic values from the task (e.g. `node.find("label").text(task.title)`), then append.

Reply:
I think this is really an underlying decision in how to update the frontend. Anther option is to send HTML back from the server that can be injected which comes with other challenges and separation issues. I think at this stage of the project it's completely fine to just generate HTML in the JS here. I don't really see a big benefit of having a unused _template_ in the original HTML that is customized with JS rather than just having the template in JS right next to the code that customizes it.
Imo if we would want to improve this we should rather look for a lightweight framework that can do this for us or maybe use Web Components. Should we pick this up on a higher level discussion in the next sync?

Reply:
The benefit of having a "template" in the original HTML is that you can see where it lands, if it matches the other HTML in the neighborhood etc.
In the end it might be reasonable to left it as-is and have a discussion in the daily sync.